collect_app/src/main/java/org/odk/collect/android/formentry/saving/FormSaveViewModel.java

```diff
@@ -370,7 +377,7 @@ public Factory(Analytics analytics) {
         @NonNull
         @Override
         public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
-            return (T) new FormSaveViewModel(System::currentTimeMillis, new DiskFormSaver(), analytics);
+            return (T) new FormSaveViewModel(System::currentTimeMillis, new DiskFormSaver(), analytics, MediaManager.INSTANCE);
```

Comment:
Maybe this hasn't come through in previous comments and conversations but I'm very against the `MediaManager` being a singleton. Just to make it very clear I think the `MediaManager` should be a `class` (not an `enum`) and that this line should be:

```java
return (T) new FormSaveViewModel(System::currentTimeMillis, new DiskFormSaver(), analytics, new MediaManager());
```

As we talked about before we could get in trouble with the `enum` singleton implementation as its state will be carried between different instances of the `FormEntryActivity`. Also in tests the singleton **would not** be recreated for each test (as the static state is retained throughout the process) so any state in the `MediaManager` set up in one test might inadvertently affect another.

Reply:
Oh right! I didn't think about it. It will cause problems when writing tests for other file widgets.